src/libs/actions/Modal.ts

```diff
@@ -1,30 +1,38 @@
 import Onyx from 'react-native-onyx';
 import ONYXKEYS from '@src/ONYXKEYS';
 
-let closeModal: (isNavigating: boolean) => void;
+const closeModals: Record<number, null | ((isNavigating: boolean) => void)> = {};
+let count = 0;
+
+// let closeModal: (isNavigating: boolean) => void;
 let onModalClose: null | (() => void);
 
+/**
+ * Get the available key that we can store the onClose callback into it
+ */
+function getAvailableKey() {
+    return count++;
+}
+
 /**
  * Allows other parts of the app to call modal close function
  */
-function setCloseModal(onClose: () => void) {
-    closeModal = onClose;
+function setCloseModal(key: number, onClose: () => void) {
```

Comment:
I've noticed that PopoverWithoutOverlay is using setCloseModal with the outdated parameter implementation, passing a function instead of a key.